source/robot_model/src/Model.cpp

```diff
@@ -299,23 +299,23 @@ Model::inverse_kinematics(const state_representation::CartesianPose& cartesian_p
   const std::chrono::nanoseconds dt(static_cast<int>(1e9));
   // initialization of the loop variables
   state_representation::JointPositions q(joint_positions);
-  state_representation::JointVelocities dq(this->get_robot_name(), this->get_number_of_joints());
-  state_representation::Jacobian J(this->get_robot_name(), this->robot_model_.nq, frame_name);
-  Eigen::MatrixXd J_b = J.data();
+  state_representation::JointVelocities dq(this->get_robot_name(), joint_positions.get_names());
+  state_representation::Jacobian J = this->compute_jacobian(joint_positions, frame_name);
+  Eigen::MatrixXd J_b =  Eigen::MatrixXd::Zero(6, this->get_number_of_joints());
```

Comment:
Here I made this change as the value of `J_b` are calculated in the loop using actually a different formula: `J_b = J * W_b` So I think it make sense to initialize it here and compute it later.